### DISCUSSION TITLE
issue with windows executable

### DISCUSSION THREAD

**Discussion by gilntk** [philosowaffle/peloton-to-garmin#124] [Q&A]
Received this error through the console after trying to download all 49 workouts and got this. Any ideas?

[log20210618.txt](https://github.com/philosowaffle/peloton-to-garmin/files/6678872/log20210618.txt)

**Comment by philosowaffle** [comment]
Scroll to the bottom of [this link](https://github.com/philosowaffle/peloton-to-garmin/actions/runs/951005361) and there is an updated windows exe you can download. Give that a try.  This has 2 updates:

1. Attempt to handle the Null error your log shows
2. If the download still fails, a json file should get added to the `output` directory, it will have more info I need to handle the issue better.

**Reply by gilntk** [reply]
(I'm a little tempted to just run the docker image, but here's the results) No luck tho:
console & output/log20210618:
```
2021-06-18 17:20:12.046 -04:00 [ERR] Failed to deserialize workout from Peloton. You can find the raw data from the workout here: @FileName
2021-06-18 17:20:12.260 -04:00 [INF] Encoded FIT file C:\Users\g\Downloads\P2G\working\0df2c80a2d0848d9a790f5668ea803fd_30_min_HIIT_&_Hills_Ride_with_Sam_Yo.fit
```
* all the individual workouts it retrieved

It did create the output/json/workout_failed_to_deserialize_1.json file that's about 11000 lines, seemingly to describe one workout. I'm a little hesitant to post it b/c of all the personal information.

output/tcx/ is empty

output/fit/ has a number of .fit files that look like they correspond to rides. But they're not text so I really can't tell you their contents.

**Reply by philosowaffle** [reply]
`output/json/workout_failed_to_deserialize_1.json `

Yup, this is the file I need. It will show me exactly which data is not coming back as expected.  Looks like I messed up the log message, instead of `@FileName` it was supposed to provide the path to that file.

[Here is an example](https://github.com/philosowaffle/peloton-to-garmin/blob/master/UnitTests/Data/3bade6602259414a82576302f9339175_workout.json#L6) workout file.  All of the identifiable information is in the first 20 lines or so.  You can scrub lines 6,14,17, and 22.  Those are the only fields that could be potentially used to identify you.  There are no emails or passwords dumped in this file I don't even think your leaderboard name will show up in there. 

The reason the file is so long is because for every second of workout it records the raw Power, HR, Cadence, and Resistance values.  There's also a lot of metadata that describes the instructor and class description. Cadence targets, Class Segments, Metric summary, it all adds up to a really lengthy file.

If docker is an option for you, I would definitely recommend that over the windows exe.

**Reply by philosowaffle** [reply]
Just a follow up note for anyone else who may stumble across this. P2G will **NEVER** intentionally print/log/display your email or password information. The only place you will ever see your email/password should be in the config file. If you ever notice P2G logging this information in plain text please open a bug so I can fix it.
